src/Inputs/Slider.tsx

```diff
@@ -1,79 +1,384 @@
+    };
+
+    // Rounding up/down to steps
+    const roundToSteps = (theValue: number): number => {
+        return (
+            Math.round(
+                ((theValue * maxAndMinDifference) /
+                    (barRef.current?.clientWidth as number) +
+                    min) /
+                    step,
+            ) * step
+        );
+    };
+
+    useLayoutEffect((): void => {
+        onChange({
+            values: {
+                leftValue: leftKnobPosition,
+                rightValue: rightKnobPosition,
+            },
+        });
+    }, [rightKnobPosition, leftKnobPosition]);
+
+    const onMouseMove = (theevent: MouseEvent): void => {
+        // clicked on pure position
+        const newLeft =
+            theevent.clientX -
+            (barRef.current as HTMLElement).getBoundingClientRect().left;
+
+        // new position based on steps
+        const newPosition = roundToSteps(newLeft);
+
+        // setting the positions
+        if (newLeft < (barRef.current?.clientWidth as number) && newLeft >= 0) {
+            if (
+                (isRightKnobDragging && newPosition > leftKnobPosition) ||
+                !hasTwoKnobs
+            ) {
+                setRightKnobPosition(newPosition);
+            } else if (isLeftKnobDragging && newPosition < rightKnobPosition) {
+                setLeftKnobPosition(newPosition);
+            }
+        }
+    };
+
+    const onMouseUp = (): void => {
+        document.removeEventListener('mouseup', onMouseUp);
+        document.removeEventListener('mousemove', onMouseMove as any);
+    };
+
+    const handleMouseDown = (e: MouseEvent): void => {
+        if (!disabled) {
+            // position the clicked taken placed on based on the position of Bar
+            const clickedOn =
+                e.clientX -
+                (barRef.current as HTMLElement).getBoundingClientRect().left;
+
+            if (
+                clickedOn >=
+                    (translateToPixels(rightKnobPosition) +
+                        translateToPixels(leftKnobPosition)) /
+                        2 ||
+                !hasTwoKnobs
+            ) {
+                setIsRightKnobDragging(true);
+                setIsLeftKnobDragging(false);
+
+                // in Case user clicks on the Bar
+                if (
+                    e.target === barRef.current ||
+                    e.target === selectedBarRef.current
+                ) {
+                    setRightKnobPosition(roundToSteps(clickedOn));
+                }
+            } else {
+                setIsRightKnobDragging(false);
+                setIsLeftKnobDragging(true);
+                // in Case user clicks on the Bar
+                if (
+                    e.target === barRef.current ||
+                    e.target === selectedBarRef.current
+                ) {
+                    setLeftKnobPosition(roundToSteps(clickedOn));
+                }
+            }
```

Comment:
You can make this code DRY by doing some thing like
 `const isOnRightSide = clickedOn >= ... || !hasTwoKnobs`
`setIsRightKnobDragging(isOnRightSide)`;
`setIsLeftKnobDragging(!isOnRightSide);`
`if (e.target ...) {isOnRightSide ? setRightKnobPosition(...) : setLeftKnob...}`

Reply:
@ryanarine its excellent...  but there is a problem on the last part, I will have to make it an assignmet like const something=isOnRightSide ? setRightKnobPosition(...) : setLeftKnob.... but I guess thats not very semantic right?

Reply:
I would just remove the ternary operator and replace it with the if-else equivalent